File: monitor-web/src/net/api-token.js
import { get, post, del } from '@/net'

/**
 * 列出当前账号的全部 API Token（不含明文）。
 *
 * @param {Function} success 成功回调，参数为 token 元数据数组
 * @param {Function} [failure] 失败回调
 */
const listTokens = (success, failure) => get('/api/tokens', success, failure)

/**
 * 创建新的 API Token。
 *
 * @param {{name: string, scope: 'readonly'|'readwrite', expiresAt?: string|null}} payload 创建请求
 * @param {Function} success 成功回调，参数为 {token: string, meta: object}；token 仅返回一次
 * @param {Function} [failure] 失败回调
 */
const createToken = (payload, success, failure) =>
  post('/api/tokens', payload, success, failure)

/**
 * 删除指定 token。
 *
 * @param {number|string} id token 主键
 * @param {Function} success 成功回调
 * @param {Function} [failure] 失败回调
 */
const deleteToken = (id, success, failure) => del(`/api/tokens/${id}`, success, failure)

/**
 * 旋转 token（删除旧 token + 按相同元数据生成新 token）。新明文同样仅返回一次。
 *
 * @param {number|string} id 旧 token 主键
 * @param {Function} success 成功回调，参数同 createToken
 * @param {Function} [failure] 失败回调
 */
const rotateToken = (id, success, failure) =>
  post(`/api/tokens/${id}/rotate`, {}, success, failure)

export { listTokens, createToken, deleteToken, rotateToken }

File: monitor-server/src/main/java/com/example/entity/dto/ApiToken.java
package com.example.entity.dto;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.util.Date;

/**
 * REST API Token 实体。对应 {@code api_token} 表。
 *
 * <p>{@code tokenHash} 列存放对完整明文 token（前缀 {@code mtk_} + 32 字符 base62 主体）计算的
 * HMAC-SHA256（URL-safe base64，无填充）。明文 token 仅在 {@code POST /api/tokens} 创建时一次性返回，
 * 不会落盘；丢失只能重新生成。
 *
 * <p>{@code prefixTail} 是 "mtk_xxxx…abcd" 形式的可识别尾部，用于列表展示但不足以重建明文。
 */
@Data
@TableName("api_token")
public class ApiToken {

    @TableId(type = IdType.AUTO)
    Long id;

    Integer accountId;

    /**
     * 用户自定义的 token 名字，用于在管理页区分用途（如 "CI/CD"、"backup script"）。
     */
    String name;

    /**
     * HMAC-SHA256(原始明文 token) 的 URL-safe base64 表示，无填充。最长 {@code VARCHAR(128)}。
     */
    String tokenHash;

    /**
     * 展示用尾部，例如 "mtk_aB3D…xY9z"（前缀 + 前 4 + … + 末 4）。
     */
    String prefixTail;

    /**
     * 二档作用域：{@code readonly} 仅放行 GET/HEAD，{@code readwrite} 放行所有方法。
     */
    String scope;

    /**
     * 过期时间；NULL 表示不过期。
     */
    Date expiresAt;

    /**
     * 最近一次成功鉴权时间。异步更新 + 60 秒节流，避免每请求一次 UPDATE。
     */
    Date lastUsedAt;

    /**
     * 最近一次成功鉴权的客户端 IP。覆盖式记录，仅保留最新一次。
     */
    String lastUsedIp;

    Date createdAt;
}
